MidiBoot3/Src/main.c

```diff
@@ -87,8 +87,8 @@ int testButton(){
 }
 
 static int testNoProgram(){  
-  return ((*(__IO uint32_t*)APPLICATION_ADDRESS) & 0x2FFE0000 ) != 0x00000000;
-  /* Check Vector Table: Test if valid stack pointer to ITCMRAM is programmed at APPLICATION_ADDRESS */
+  return ((*(__IO uint32_t*)APPLICATION_ADDRESS) & 0x2FFE0000 ) != 0x20000000;

```

Comment:
Let's check that the Reset_Handler entry point is valid instead, something like
```((*(__IO uint32_t*)APPLICATION_ADDRESS+1) & 0xfff00000) != 0x08000000```

Reply:
In such case I'll update other bootloaders to do the same for consistency.

Btw, for Daisy there's no need to perform such ugly checks, VTOR address is taken from firmware header directly. So it's possible to change without altering bootloader, we only have to confirm that firmware header magic is found where expected. And that header is generated automatically while compiling, based on linker exports.

Reply:
Very funny, your expression is checking address 0x20000001. The joy of operator precedence and associativity rules!

I've changed it slightly and tested with owl2 bootloader.

Reply:
Just keeping you on your toes!